toponetx/classes/path.py

```diff
@@ -0,0 +1,120 @@
+"""Path class."""
+
+from collections.abc import Collection, Hashable
+from typing import Any, List, Tuple, Union
+
+from toponetx.classes.complex import Atom
+
+__all__ = ["Path"]
+
+
+class Path(Atom):
+    """
+    A class representing a p-path in a path complex, which is the building block for a path complex.
+
+    Unlike the original paper (https://arxiv.org/pdf/1207.2834.pdf) where p-paths span the regular space of boundary-invariant paths,
+    our p-paths span the space of simple paths with length p.
+
+    Parameters
+    ----------
+    elements: Collection
+        The nodes in the p-path.
+    name : str, optional
+        A name for the p-path.
+    construct_boundaries : bool, default=False
+        If True, construct the entire boundary of the p-path.
+    reserve_sequence_order : bool, default=False
+        If True, reserve the order of the sub-sequence of nodes in the p-path.
+        Else, the sub-sequence of nodes in the p-path will be reversed if the first index is larger than the last index.
+    allowed_paths : List[Tuple], optional
+        A list of allowed boundaries. If None, all possible boundaries are constructed (similarly to simplex).
+    attr: keyword arguments, optional
+        Additional attributes to be associated with the p-path.
+    """
+
+    def __init__(
+        self,
+        elements: Union[List, Tuple],
```

Comment:
The type definition is insufficient:

1. What are the elements of the list or tuple? I suspect `Hashable`? (This applies throughout the pull request)
2. Do you really need a list or a tuple? Probably a protocol from `collections.abc` would be more fitting.

Reply:
1. Yes it should be `Hashable`.
2. I think `Sequence` is appropriate in this case because sequence order matters in path.